### PR TITLE
MVP envvar container

### PR DESCRIPTION
**Pull request by yapishu**
Takes a pubkey as env var and creates user. Edit .env with the pubkey, and the template client config with the endpoint address and client privkey. Runs on 51820/UDP

### CODE REVIEW

**Comment by yapishu** [comment]
sorry for the delay on this, i broke my computer last weekend and didn't get to it when i was planning

File: buildmod.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+set -euo pipefail
+
+if [ -f /usr/src/app/wireguard.ko ]
+then
+	modinfo /usr/src/app/wireguard.ko
+	echo "wireguard.ko already exists, skipping module build..."
+	exit 0
+fi
+
+if [ -n "${BALENA_DEVICE_TYPE:-}" ]
```
Comment (odyslam) [inline buildmod.sh]: Why do we want this? The server will run on the cloud

**Comment by odyslam** [comment]
@yapishu I have left some comments. Has this setup been tested with the anchor branch of home-urbit ? 

Ideally, we should have a PR in that repo as well with a setup that works with the PR here. 

The PR here is meant to be executed just by running `docker-compose up` ? Have we verified that it works in a Cloud VM environments? (There are some strange things that can happen with ingress/egress traffic and iptables).

**Comment by yapishu** [comment]
@odyslam yes, i tested it on EC2 with `docker-compose up` + the pubkey envvar, but I haven't tested it from the anchor branch, just from my PC. i'll go ahead and plug it into that tomorrow but i don't anticipate any issues, wireguard is very straightforward comparatively

**Comment by odyslam** [comment]
@yapishu can you also remove all the balena mentions, since this container is meant to be run in the cloud? 

Also, I remember that I had problems with the iptables, as they would interact with the underlying virtualization OS and I would get metered for the network of the entire physical machine, not my VM. Can you verify that this is not the case with the cfonig heere.

**Comment by yapishu** [comment]
@odyslam here's home container-to-server container: 
image

the client host device also does not report the IP of the server when it's not inside the container, so it's not tunneling all host traffic through:
image

the caveat is that the home-urbit container is set up to be a server, not a client, by default, so i had to pop inside to manually turn off the default wg interface and enable the client interface (obviously pretty easy to script in)

**Review by odyslam — APPROVED** [review]
LGTM

Final nit: Let's make this a `docker run` one-liner vs using docker-compose. Since it's a single container, we add an unnecessary complexity and dependency

**Pull request merged**